### ISSUE TITLE
OTX2.0 Anomaly tasks show 0 accuracy on mvtec datasets

### ISSUE THREAD

**Issue by goodsong81** [open-edge-platform/training_extensions#3219]
**Describe the bug**
![image](https://github.com/openvinotoolkit/training_extensions/assets/4629606/f16b4e4d-bc84-47ed-9560-d10342fa457a)

**Steps to Reproduce**

On the latest releases/2.0.0 branch
```bash
otx train --config src/otx/recipe/anomaly_classification/padim.yaml --data_root /mnt/sdc/val-data/v1/anomaly/mvtec/wood_medium/ --work_dir /tmp/ano
otx test --work_dir /tmp/ano
```
Same goes for stfpm and other mvtec datasets.

**Comment by ashwinvaidya17** [comment]
Yeah it is a known issue. This is due to the fact that the default dataset configured from the CLI does not contain validation set. So the thresholds are not assigned correctly. Ideally OTX should support custom dataloaders.

I've documented this in the config files
https://github.com/openvinotoolkit/training_extensions/blob/2277ec5fa01330a1279ebb07be742c29722369b2/src/otx/recipe/anomaly_classification/padim.yaml#L19

If we use this https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/data/anomaly/anomaly.py from the API then it gives good numbers

**Comment by harimkang** [comment]
> Yeah it is a known issue. This is due to the fact that the default dataset configured from the CLI does not contain validation set. So the thresholds are not assigned correctly. Ideally OTX should support custom dataloaders.
> 
> I've documented this in the config files
> 
> https://github.com/openvinotoolkit/training_extensions/blob/2277ec5fa01330a1279ebb07be742c29722369b2/src/otx/recipe/anomaly_classification/padim.yaml#L19
> 
> If we use this https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/data/anomaly/anomaly.py from the API then it gives good numbers

@ashwinvaidya17 Is there any workaround for this?

**Comment by harimkang** [comment]
https://github.com/openvinotoolkit/training_extensions/pull/3231 Merged into 2.0
